docs/structure-of-a-contract.rst

```diff
@@ -42,6 +42,41 @@ Functions are the executable units of code within a contract.
 and have different levels of visibility (:ref:`visibility-and-getters`)
 towards other contracts. Functions must be decorated with either @public or @private.
 
+Default function
+----------------
+
+A contract can also have a default function, which is executed on a call to the contract if no other functions match the given function identifier (or if none was supplied at all, such as through someone sending it Eth). It is the same construct as fallback functions `in Solidity <https://solidity.readthedocs.io/en/latest/contracts.html?highlight=fallback#fallback-function>`_. 
```

Comment:
"if no other functions match the given function identifier" I am pretty sure this is the behavior we have, but I don't actually like this behavior now that I think about it. The default function should not have any call data associated with it (aka fails if call data is provided) and I think in general if the function signature switch falls through, the contract should revert by default. We can keep this behavior in place for the fallback function, since that will check that call data is zero if no function signatures match.

@jacqueswww Please confirm current behavior matches this documentation, and also comment on my suggestion for modifying the behavior of the default function. This should be compatible with any contracts created by Solidity. In fact, do we have a test case for one contract sending another Eth internally?

Reply:
@fubuloubu this is the current behaviour.

And therefore I think it's fine to leave it as is (in the docs). However we can definitely scope a VIP for splitting default function into 2 cases:
1.) Receive funds 
2.) Function Not Found (404)
I saw this on solidity something similar is being discussed. https://github.com/ethereum/solidity/issues/3198#issuecomment-380493711